File: src/SimpleStocker.Web/src/app/models/saleItem.model.ts
import { BaseModel } from './basemodel.model';

export class SaleItem extends BaseModel {
  saleId: number = 0;
  productId: number = 0;
  quantity: number = 0;
  unityPrice: number = 0;
  subTotal: number = 0;

  constructor() {
    super();
    this.saleId = 0;
    this.productId = 0;
    this.quantity = 0;
    this.unityPrice = 0;
    this.subTotal = 0;
  }
}

File: src/SimpleStocker.SaleApi/Models/SaleItemModel.cs
using SimpleStocker.Api.Models.Entities;

namespace SimpleStocker.SaleApi.Models.Entities
{
    public class SaleItemModel
    {
        public long Id { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime UpdatedDate { get; set; }
        public SaleModel Sale { get; set; }
        public long SaleId { get; set; }
        public long ProductId { get; set; }
        public double Quantity { get; set; }
        public decimal UnityPrice { get; set; }
        public decimal SubTotal { get; set; }
    }
}
